src/ops/fuelup_toolchain/new.rs

```diff
@@ -0,0 +1,40 @@
+use crate::path::{ensure_dir_exists, settings_file, toolchain_bin_dir};
+use crate::settings::SettingsFile;
+use crate::{commands::toolchain::NewCommand, path::toolchain_dir};
+use anyhow::bail;
+use anyhow::Result;
+use std::fs;
+use std::io;
+
+pub fn new(command: NewCommand) -> Result<()> {
+    let NewCommand { name } = command;
+
+    let toolchain_dir = toolchain_dir();
+
+    let toolchain_exists = fs::read_dir(&toolchain_dir)?
+        .filter_map(io::Result::ok)
+        .filter(|e| e.file_type().map(|f| f.is_dir()).unwrap_or(false))
```

Comment:
I think this could also be:
```suggestion
        .filter(|e| e.path().is_dir())
```